tools/tessolarcharge.py

```diff
@@ -0,0 +1,272 @@
+"""
+ Adapted by Nate Carroll 03/2023 from TesSense w/ SenseLink  -Randy Spencer 2023 Version 9.7
+ Python charge monitoring utility for those who have a Tesla Powerwall.
+ Uses jasonacox/pypowerwall docker container to get realtime stats from the local gateway for Production 
+ and Grid Utilization of electricity to control
+ your main Tesla's AC charging amps to charge only with excess production.
+ Simply plug in your car, update your info below, and type> python3 tessolarcharge.py
+ 
+ Added: checking of location of Tesla to be sure it's charging at home
+ Added: tracks cabin temp and local chargers, vents the car if it gets too hot 
+ - removed due to Tesla removal of Vent option from API
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND ON INFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+
+username = 'elon@tesla.com' # Fill in Tesla login email address/account
```

Comment:
I'm sure Elon would appreciate if we just used the ICANN approved example address "yourname@example.com" instead :)

Reply:
Ha ha, maybe if everyone charged his car with their solar he wouldn't need all of that Twitter verified money? But good idea, I'll change and submit another PR. :-)

Reply:
Thanks!  You won't need to submit another PR.  Just commit and push your changes to the branch you used for this PR (`venturanc:main`).